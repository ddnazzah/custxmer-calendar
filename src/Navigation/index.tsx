/*
 * @Author: Dieu-Donne Nazzah
 * @Date: 2022-10-24 04:20:57
 * @Last Modified by: Dieu-Donne Nazzah
 * @Last Modified time: 2022-10-24 05:02:33
 */
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { Constants } from '@shared/constants';
import React from 'react';
import { SafeAreaView } from 'react-native-safe-area-context';
import { MyCalendar } from '../MyCalendar';
import { Splash } from '../Splash';

const MainNavigation = createNativeStackNavigator();

const routes = [
    {
        name: Constants.routes.SPLASH,
        component: Splash,
    },
    {
        name: Constants.routes.CALENDAR,
        component: MyCalendar,
    },
];

const Navigator = () => {
    return (
        <SafeAreaView style={{ flex: 1 }}>
            <NavigationContainer>
                <MainNavigation.Navigator
                    screenOptions={{ headerShown: false }}>
                    {routes.map(route => (
                        <MainNavigation.Screen
                            key={route.name}
                            name={route.name}
                            component={route.component}
                        />
                    ))}
                </MainNavigation.Navigator>
            </NavigationContainer>
        </SafeAreaView>
    );
};

export default Navigator;
