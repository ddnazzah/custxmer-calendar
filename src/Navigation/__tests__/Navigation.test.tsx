import { render, screen } from '@testing-library/react-native';
import React from 'react';
import Navigation from '..';
import TestProvider from '../../@testing/Provider';

describe('Navigation', () => {
    it('renders correctly', () => {
        render(
            <TestProvider>
                <Navigation />
            </TestProvider>,
        );

        expect(screen.toJSON()).toMatchSnapshot();
    });
});
