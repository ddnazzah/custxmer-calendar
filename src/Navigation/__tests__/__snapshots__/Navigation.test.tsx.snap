// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Splash"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <BVLinearGradient
            colors={
              Array [
                4278190080,
                4294400067,
                4279311535,
              ]
            }
            endPoint={
              Object {
                "x": 0.5,
                "y": 1,
              }
            }
            locations={null}
            startPoint={
              Object {
                "x": 0.5,
                "y": 0,
              }
            }
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 36,
                    "letterSpacing": 1,
                  },
                ]
              }
            >
              Custxmer
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 36,
                    "letterSpacing": 1,
                  },
                ]
              }
            >
              Calendar
            </Text>
          </BVLinearGradient>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
